x-pack/plugins/apm/public/components/app/transaction_details/index.tsx

```diff
@@ -69,16 +74,25 @@ export function TransactionDetails() {
       <EuiSpacer size="m" />
 
       <ChartPointerEventContextProvider>
-        <TransactionCharts
-          kuery={query.kuery}
-          environment={query.environment}
-          start={start}
-          end={end}
-          transactionName={transactionName}
-          isServerlessContext={isServerless}
-          comparisonEnabled={comparisonEnabled}
-          offset={offset}
-        />
+        {!isMobileAgent ? (
+          <TransactionCharts
+            kuery={query.kuery}
+            environment={query.environment}
+            start={start}
+            end={end}
+            transactionName={transactionName}
+            isServerlessContext={isServerless}
+            comparisonEnabled={comparisonEnabled}
+            offset={offset}
+          />
+        ) : (
+          <MobileAgentTransactionCharts
+            kuery={query.kuery}
+            environment={query.environment}
+            start={start}
+            end={end}
+          />
+        )}
```

Comment:
```suggestion
        {isMobileAgent ? (
           <MobileAgentTransactionCharts
            kuery={query.kuery}
            environment={query.environment}
            start={start}
            end={end}
          />
        ) : (
        <TransactionCharts
            kuery={query.kuery}
            environment={query.environment}
            start={start}
            end={end}
            transactionName={transactionName}
            isServerlessContext={isServerless}
            comparisonEnabled={comparisonEnabled}
            offset={offset}
          />
        )}
```

Reply:
I think that it makes sense to have the more generic component first but updated anyways in https://github.com/elastic/kibana/pull/143697/commits/9cab05301c4352db61467fd0923f573308443f48